tensorflow/core/kernels/data/batch_dataset_op_test.cc

```diff
@@ -109,96 +58,106 @@ class BatchDatasetOpTest : public DatasetOpsTestBaseV2<BatchDatasetParams> {
 // Test Case 1: test BatchDatasetV2 with `drop_remainder` = false and a batch
 // size that can evenly split the input dataset.
 BatchDatasetParams BatchDatasetParams1() {
-  return {/*num_input_elements=*/12,
-          /*batch_size=*/4,
-          /*drop_remainder=*/false,
-          /*parallel_copy=*/true,
-          /*output_dtypes=*/{DT_INT64},
-          /*output_shapes=*/{PartialTensorShape({4})},
-          /*node_name=*/kNodeName};
+  auto range_dataset_params = RangeDatasetParams(0, 12, 1);
+  return BatchDatasetParams(std::move(range_dataset_params),
+               /*batch_size=*/4,
+               /*drop_remainder=*/false,
+               /*parallel_copy=*/true,
+               /*output_dtypes=*/{DT_INT64},
+               /*output_shapes=*/{PartialTensorShape({4})},
+               /*node_name=*/kNodeName);
 }
 
 // Test Case 2: test BatchDatasetV2 with `drop_remainder` = true and a batch
 // size that can evenly split the input dataset.
 BatchDatasetParams BatchDatasetParams2() {
-  return {/*num_input_elements=*/12,
-          /*batch_size=*/4,
-          /*drop_remainder=*/true,
-          /*parallel_copy=*/false,
-          /*output_dtypes=*/{DT_INT64},
-          /*output_shapes=*/{PartialTensorShape({4})},
-          /*node_name=*/kNodeName};
+  auto range_dataset_params = RangeDatasetParams(0, 12, 1);
+  return BatchDatasetParams(std::move(range_dataset_params),
+               /*batch_size=*/4,
+               /*drop_remainder=*/true,
+               /*parallel_copy=*/false,
+               /*output_dtypes=*/{DT_INT64},
+               /*output_shapes=*/{PartialTensorShape({4})},
+               /*node_name=*/kNodeName);
 }
 
 // Test Case 3: test BatchDatasetV2 with `drop_remainder` = false and a batch
 // size that can not evenly split the input dataset.
 BatchDatasetParams BatchDatasetParams3() {
-  return {/*num_input_elements=*/10,
-          /*batch_size=*/3,
-          /*drop_remainder=*/false,
-          /*parallel_copy=*/false,
-          /*output_dtypes=*/{DT_INT64},
-          /*output_shapes=*/{PartialTensorShape({-1})},
-          /*node_name=*/kNodeName};
+  auto range_dataset_params = RangeDatasetParams(0, 10, 1);
+  auto batch_dataset_params_0 =
```

Comment:
no need to create a variable for this one, you can just return it

Reply:
Yeah, changed it.